devito/finite_differences/finite_difference.py

```diff
@@ -64,27 +63,33 @@ def wrapper(expr, *args, **kwargs):
 
 
 @check_input
-def first_derivative(expr, **kwargs):
+def first_derivative(expr, dim, fd_order, side=centered, matvec=direct):
     """
     First-order derivative of a given expression.
 
     Parameters
     ----------
     expr : expr-like
         Expression for which the first-order derivative is produced.
-    **kwargs
-        - ``dim``: Dimension w.r.t. which to differentiate.
-        - ``diff``: Finite-difference symbol, defaults to `h`.
-        - ``order``: Discretisation order of the derivative.
-
+    dim : Dimension
+        The Dimension w.r.t. which to differentiate.
+    fd_order : int
+        Coefficient discretization order. Note: this impacts the width of
+        the resulting stencil.
+    side : Side
```

Comment:
why is this not in `second_derivative` and all other functions?

Reply:
, optional

Reply:
Because second derivative is "dense" and doesn't need tricks to have value at all grid points (first derivative -s [-1, 0, 1]` while second is [-1, 2, -1]` no 0 in it)